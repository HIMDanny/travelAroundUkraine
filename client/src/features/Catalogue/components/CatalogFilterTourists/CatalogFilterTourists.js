import React from 'react';
import Grid from '@mui/material/Unstable_Grid2';
import { FilterAccordion, OutlinedToggleButton } from '..';

const touristsNumber = Array(10)
  .fill()
  .map((_, i) => i + 1);

const CatalogFilterTourists = () => (
  <FilterAccordion title="Tourists">
    <Grid container spacing={1}>
      {touristsNumber.map((el) => (
        <Grid xs={4} key={el}>
          <OutlinedToggleButton value={el}>{el} pers.</OutlinedToggleButton>
        </Grid>
      ))}

      <Grid xs={8}>
        <OutlinedToggleButton>More than 10 pers.</OutlinedToggleButton>
      </Grid>
    </Grid>
  </FilterAccordion>
);

export default CatalogFilterTourists;
