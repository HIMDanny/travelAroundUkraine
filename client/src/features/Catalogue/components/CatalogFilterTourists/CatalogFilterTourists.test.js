import { render } from '@testing-library/react';
import CatalogFilterTourists from './CatalogFilterTourists';

jest.mock('@mui/material/ToggleButton', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Unstable_Grid2', () => ({ children }) => <div>{children}</div>);

jest.mock('../FilterAccordion/FilterAccordion', () => ({ children }) => <div>{children}</div>);
jest.mock('../OutlinedToggleButton/OutlinedToggleButton', () => () => <button>test button</button>);

describe('CatalogFilterTourists snapshot test', () => {
  test('should CatalogFilterTourists match snapshot', () => {
    const { asFragment } = render(<CatalogFilterTourists />);
    expect(asFragment()).toMatchSnapshot();
  });
});
