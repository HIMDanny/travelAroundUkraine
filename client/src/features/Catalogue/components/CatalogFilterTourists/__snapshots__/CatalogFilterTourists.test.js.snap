// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogFilterTourists snapshot test should CatalogFilterTourists match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
      <div>
        <button>
          test button
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
